laravel/app/Http/routes.php

```diff
@@ -111,6 +111,7 @@
 Route::get('/users', 'AdminController@userList');
 Route::get('/user/{user}', 'AdminController@userProfile');
 Route::post('/user/{user}/edit', 'AdminController@userEdit');
+Route::post('/user/{user}/update', 'AdminController@userProfileUpdate');
```

Comment:
Why are there now two separate post requests for `userEdit` and `userProfileUpdate`? What is the difference between these two endpoints?

Reply:
One is hit via ajax when you only update one field. I thought maybe it would be simpler but can consolidate.

Reply:
It seems the button that was clicked to hit the edit route and update just roles (`$('.save-roles')`) no longer exists anyway. see recent commit for more info